core/src/main/java/jenkins/util/AtmostOneTaskExecutor.java

```diff
@@ -100,6 +107,7 @@ public Void call() throws Exception {
                     try {
                         inprogress.set(task.call());
                     } catch (Throwable t) {
+                        LOGGER.log(Level.WARNING, null, t);
                         inprogress.setException(t);
```

Comment:
…because this has no effect whatsoever. I do not really understand what @kohsuke was trying to accomplish in 666f0b47816f55a3310cee7d5da0d42b8e354e5a, but the `inprogress` field is only ever modified or checked for nullity—meaning it could just as well have been a `boolean`, as far as I can tell.